zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/usertask/UserTaskRecord.java

```diff
@@ -459,6 +509,33 @@ public UserTaskRecord setChangedAttributesProp(
     return this;
   }
 
+  public UserTaskRecord addChangedAttribute(final String attribute) {
+    changedAttributesProp.add().wrap(BufferUtil.wrapString(attribute));
+    return this;
+  }
+
+  public void setDiffAsChangedAttributes(final UserTaskRecord other) {
+    changedAttributesProp.reset();
+    if (!getAssigneeBuffer().equals(other.getAssigneeBuffer())) {
+      addChangedAttribute(ASSIGNEE);
+    }
+    if (!getCandidateGroupsList().equals(other.getCandidateGroupsList())) {
+      addChangedAttribute(CANDIDATE_GROUPS);
+    }
+    if (!getCandidateUsersList().equals(other.getCandidateUsersList())) {
+      addChangedAttribute(CANDIDATE_USERS);
+    }
+    if (!getDueDateBuffer().equals(other.getDueDateBuffer())) {
+      addChangedAttribute(DUE_DATE);
+    }
+    if (!getFollowUpDateBuffer().equals(other.getFollowUpDateBuffer())) {
+      addChangedAttribute(FOLLOW_UP_DATE);
+    }
+    if (getPriority() != other.getPriority()) {
+      addChangedAttribute(PRIORITY);
+    }
+  }
```

Comment:
:thought_balloon: **Idea:**

To enhance readability and simplify maintenance, we can use a helper method to replace the repetitive `if` statements. This way, the code would be more concise and makes it easier to add future comparisons.

For example:

```java
public void setDiffAsChangedAttributes(final UserTaskRecord other) {
  changedAttributesProp.reset();
  addIfAttributeChanged(ASSIGNEE, this::getAssigneeBuffer, other::getAssigneeBuffer);
  addIfAttributeChanged(CANDIDATE_GROUPS, this::getCandidateGroupsList, other::getCandidateGroupsList);
  addIfAttributeChanged(CANDIDATE_USERS, this::getCandidateUsersList, other::getCandidateUsersList);
  addIfAttributeChanged(DUE_DATE, this::getDueDateBuffer, other::getDueDateBuffer);
  addIfAttributeChanged(FOLLOW_UP_DATE, this::getFollowUpDateBuffer, other::getFollowUpDateBuffer);
  addIfAttributeChanged(PRIORITY, this::getPriority, other::getPriority);
}

private <T> void addIfAttributeChanged(
    final String attribute, final Supplier<T> currentValue, final Supplier<T> otherValue) {
  if (ObjectUtils.notEqual(currentValue.get(), otherValue.get())) {
    addChangedAttribute(attribute);
  }
}
```

Alternatively, to make code more concise and reduce the risk of mixing up different properties during comparison:

```java
public void setDiffAsChangedAttributes(final UserTaskRecord other) {
  changedAttributesProp.reset();
  addIfAttributeChanged(ASSIGNEE, UserTaskRecord::getAssigneeBuffer, other);
  addIfAttributeChanged(CANDIDATE_GROUPS, UserTaskRecord::getCandidateGroupsList, other);
  addIfAttributeChanged(CANDIDATE_USERS, UserTaskRecord::getCandidateUsersList, other);
  addIfAttributeChanged(DUE_DATE, UserTaskRecord::getDueDateBuffer, other);
  addIfAttributeChanged(FOLLOW_UP_DATE, UserTaskRecord::getFollowUpDateBuffer, other);
  addIfAttributeChanged(PRIORITY, UserTaskRecord::getPriority, other);
}

private <T> void addIfAttributeChanged(
    final String attribute,
    final Function<UserTaskRecord, T> attributeGetter,
    final UserTaskRecord other) {
  if (ObjectUtils.notEqual(attributeGetter.apply(this), attributeGetter.apply(other))) {
    addChangedAttribute(attribute);
  }
}
```

Just an idea for consideration - feel free to adopt it if you find it useful! 😊

Reply:
This is a great idea! Love it!

Reply:
I've taken your idea and made two small changes:
- only use java.util methods (`!Objects.equals()` rather than `ObjectUtils.notEqual`)
- place `attributeGetter` results in local variables for easier debugging and readability